app/routes/modern-web-podcast/index.js

```diff
@@ -1,15 +1,64 @@
 import Ember from 'ember';
 
-const ensureImage = (project) => (podcast) => project({
-  ...podcast,
+
+/*
+NOTES FOR TRACY
+=========================
+
+1. What is Object.assign? It's a way to copy an properties from a list of objects onto another object.
+   So if you do `Object.assign(a, b, c)` it's going to take all of the properties from b and c and copy them
+   onto `a`, then return `a`. If you want to duplicate an object, so you don't alter the original object, you'd
+   do: `const copy = Object.assign({}, original);`
+2. This uses "functional programming" an higher-order functions in a fairly basic way. However, these techniques
+   can be confusing even for advanced developers. So if it looks weird to you, take heart, it looks weird to 90%
+   of developers too, for the most part.
+*/
+
+/**
+  higher-order function to ensure there's an image on each podcast in a list. It's a "higher-order"
+  function because it's a function that returns a function.
+
+  This looks funny, because it's a function written so you can chain it with other functions.
+  That way we can loop through the podcasts _only once_ with a map (below), and fix them
+  all in one go.
+
+  @param {function} project - This is a projection function, a projection function is any function that
+    can be used with an Array `map`. It projects a value from the array into a new value.
+
+  @returns {function} a projection function! When you call `ensureImage` it's going to take a function
+    you can use in an Array `map`, and return a function you can use in an Array `map`. It chains two
+    transformations together, so you you only have to `map` once.
+*/
+const ensureImage = (project) => (podcast) => project(Object.assign({}, podcast, {
   imageURL: podcast.imageURL || `http://img.youtube.com/vi/${podcast.id}/hqdefault.jpg`
-});
+}));
 
-const splitDescriptionParagraphs = (project) => (podcast) => project({
-  ...podcast,
+/**
+  An ordinary projection function that copies a podcast object onto a new one, but adds the list of
+  descriptionParagraphs
+  @param {object} podcast the podcast you want to add the descriptionParagraphs to
+  @returns {object} a new podcast object with the descriptionParagraphs property added.
+*/
+const splitDescriptionParagraphs = (podcast) => Object.assign({}, podcast, {
   descriptionParagraphs: podcast.description.split('\n')
```

Comment:
since description is required to be present in the podcast object we should annotate the param as such since if description is not present this code will bail